Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -356,185 +472,304 @@ void LoadPSIMuonBin::assignOutputWorkspaceParticulars(
+                     m_header.firstGood[i], outputWorkspace);
+      addToSampleLog("Last good spectra " + std::to_string(i),
+                     m_header.lastGood[i], outputWorkspace);
     }
   }
 
-  // total events
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Total Number of Events");
-  logAlg->setProperty("LogText", std::to_string(m_header.totalEvents));
-  logAlg->executeAsChildAlg();
-
-  // tdcResolution
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "TDC Resolution");
-  logAlg->setProperty("LogText", std::to_string(m_header.tdcResolution));
-  logAlg->executeAsChildAlg();
-
-  // tdcOverflow
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "TDC Overflow");
-  logAlg->setProperty("LogText", std::to_string(m_header.tdcOverflow));
-  logAlg->executeAsChildAlg();
-
-  // Length of spectra
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Spectra Length");
-  logAlg->setProperty("LogText", std::to_string(m_header.lengthOfHistograms));
-  logAlg->executeAsChildAlg();
-
-  // Number of Spectra
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Number of Spectra");
-  logAlg->setProperty("LogText", std::to_string(m_header.numberOfHistograms));
-  logAlg->executeAsChildAlg();
-
-  // monNumber of events
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Mon number of events");
-  logAlg->setProperty("LogText", std::to_string(m_header.monNumberOfevents));
-  logAlg->executeAsChildAlg();
-
-  // Mon Period
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Mon Period");
-  logAlg->setProperty("LogText", std::to_string(m_header.periodOfMon));
-  logAlg->executeAsChildAlg();
+  addToSampleLog("TDC Resolution", m_header.tdcResolution, outputWorkspace);
+  addToSampleLog("TDC Overflow", m_header.tdcOverflow, outputWorkspace);
+  addToSampleLog("Spectra Length", m_header.lengthOfHistograms,
+                 outputWorkspace);
+  addToSampleLog("Number of Spectra", m_header.numberOfHistograms,
+                 outputWorkspace);
+  addToSampleLog("Mon number of events", m_header.monNumberOfevents,
+                 outputWorkspace);
+  addToSampleLog("Mon Period", m_header.periodOfMon, outputWorkspace);
 
   if (m_header.monLow[0] == 0 && m_header.monHigh[0] == 0) {
-    logAlg->setProperty("LogType", "String");
-    logAlg->setProperty("LogName", "Mon Low");
-    logAlg->setProperty("LogText", "0");
-    logAlg->executeAsChildAlg();
-
-    logAlg->setProperty("LogType", "String");
-    logAlg->setProperty("LogName", "Mon High");
-    logAlg->setProperty("LogText", "0");
-    logAlg->executeAsChildAlg();
+    addToSampleLog("Mon Low", "0", outputWorkspace);
+    addToSampleLog("Mon High", "0", outputWorkspace);
   } else {
     for (auto i = 0u; i < 4; ++i) {
       if (m_header.monLow[i] == 0 || m_header.monHigh[i] == 0)
```

Comment:
would it be better to put the if (and reverse the conditions) to include the for?

Reply:
In that case, the for loop would still iterate over the if statement every time and evaluate false, in this case, it stops executing the loop at that point, making it more efficient and in my opinion more explicit.

To achieve the same thing I would write this code:
![image](https://user-images.githubusercontent.com/24723609/57512097-ab257b80-7302-11e9-8ac2-fd61d8a78482.png)